src/control-plane/backend/lambda/api/common/utils.ts

```diff
@@ -106,6 +110,66 @@ function getEmailFromRequestContext(requestContext: string | undefined) {
   return email;
 }
 
+function getTokenFromRequestContext(requestContext: string | undefined) {
+  let token = '';
+  try {
+    if (requestContext) {
+      // Api Gateway pass the request context to the backend
+      const context = JSON.parse(requestContext);
+      token = context.authorizer.authorizationToken ?? '';
+    }
+  } catch (err) {
+    logger.warn('unknown user', {
+      requestContext,
+      err,
+    });
+  }
+  return token;
+}
+
+function getTokenFromRequest(req: any) {
+  let authorization;
+  const WITH_AUTH_MIDDLEWARE = process.env.WITH_AUTH_MIDDLEWARE;
+  if (WITH_AUTH_MIDDLEWARE === 'true') {
+    authorization = req.get('authorization');
+  } else {
+    authorization = getTokenFromRequestContext(req.get(amznRequestContextHeader));
+  }
+  if (authorization) {
+    const token = authorization.split(' ')[1];
+    const decodedToken = jwt.decode(token, { complete: true });
+
+    return decodedToken;
+  }
+  return undefined;
+}
+
+function getRoleFromToken(decodedToken: any) {
+  let role = IUserRole.NO_IDENTITY;
+  if (!decodedToken) {
+    return role;
+  }
+
+  let oidcRoles: string[] = [];
+
+  const OIDC_ROLE_PATH = process.env.OIDC_ROLE_PATH ?? '$.payload.cognito:groups';
+  const values = JSONPath({ path: OIDC_ROLE_PATH, json: decodedToken });
+  if (Array.prototype.isPrototypeOf(values) && values.length > 0) {
+    oidcRoles = values[0] as string[];
+  } else {
+    return role;
+  }
+  if (oidcRoles &&
+          oidcRoles.includes(`Clickstream${IUserRole.OPERATOR}`) &&
+          oidcRoles.includes(`Clickstream${IUserRole.ANALYST}`)) {
```

Comment:
Let's move it configurable via env like `token path`.

Reply:
use [this](https://github.com/awslabs/clickstream-analytics-on-aws/pull/205/files#diff-720788bdfb4c591afcdd0e55c0377f7354aaf58151887edc3db3250191c3a246R155) read groups or roles from token